### PR TITLE
expression: implement vectorized evaluation for `builtinQuarterSig`

### PR DESCRIPTION
**Pull request by 710698117**
### What problem does this PR solve? 
implement vectorized evaluation for `builtinQuarterSig` , for #12103

### What is changed and how it works?
```
go test -v -benchmem -bench=BenchmarkVectorizedBuiltinTimeFunc -run=BenchmarkVectorizedBuiltinTimeFunc
goos: linux
goarch: amd64
pkg: github.com/pingcap/tidb/expression
BenchmarkVectorizedBuiltinTimeFunc/builtinQuarterSig-VecBuiltinFunc-4                                 	   50000	     25491 ns/op	       0 B/op	       0 allocs/op
BenchmarkVectorizedBuiltinTimeFunc/builtinQuarterSig-NonVecBuiltinFunc-4                              	   30000	     57797 ns/op	       0 B/op	       0 allocs/op
```

### Check List 

Tests 

 - Unit test

### CODE REVIEW

File: expression/builtin_time_vec.go [hunk 1]
```diff
@@ -541,11 +541,39 @@ func (b *builtinSubDateStringStringSig) vecEvalTime(input *chunk.Chunk, result *
...
+	n := input.NumRows()
+	buf, err := b.bufAllocator.get(types.ETDatetime, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(buf)
+	if err := b.args[0].VecEvalTime(b.ctx, input, buf); err != nil {
+		if err := handleInvalidTimeError(b.ctx, err); err != nil {
+			return err
+		}
```
Comment (SunRunAway) [inline expression/builtin_time_vec.go]: We need not call `handleInvalidTimeError` here, it's `args[0].VecEvalTime`'s responsibility to handle time error line by line.

Comment (710698117) [reply expression/builtin_time_vec.go]: Thanks, I' ve fix it.

**Comment by SunRunAway** [comment]
Hi, @710698117 
Thanks for your contribution, your first PR looks awesome,
Unfortunately there's a PR https://github.com/pingcap/tidb/pull/13089 taking the same function.
I'll close yours because that is earlier than yours. Sorry about that.

You can choose another vectorized job to work. Looking forward to your next PR.
Sorry again.

**Comment by SunRunAway** [comment]
Duplicate of #13089

**Comment by 710698117** [comment]
> Duplicate of #13089



> Duplicate of #13089

Oh my god, for the second today :( #13105

**Comment by SunRunAway** [comment]
Sorry about that. You may want to search before coding if there’s a duplication existing.

**Comment by 710698117** [comment]
> Sorry about that. You may want to search before coding if there’s a duplication existing.

Ok, thanks for informing~

**Pull request closed without merging**